src/main/webapp/index-script.js

```diff
@@ -82,22 +85,30 @@ function searchByCoordinates(coordinate) {
 }
 
 // Search using text query
-function searchByText(){
+function searchByText(textQuery){
   // Create the places service.
-  const service = new google.maps.places.PlacesService(map);
-  
+
   // Perform a query (hard-coded to be the Googleplex for right now)
   var request = {
-      query: 'Googleplex',
-      radius: '1000',
+      query: textQuery,
       fields: ['place_id', 'geometry']
-  }
+  };
+
+  var service = new google.maps.places.PlacesService(map);
 
   service.findPlaceFromQuery(request, function(results, status) {
     if (status === google.maps.places.PlacesServiceStatus.OK) {
-      for (var i = 0; i < results.length; i++){
-          // Populate the screen
-      }
+      results.forEach((result)=> {
+        var request = {
+          placeId: result.place_id,
+          fields: ['name']
+        };
+        service.getDetails(request, (place, status) => {
+          if (status == google.maps.places.PlacesServiceStatus.OK) {
+            console.log(place.name);
+          }
+        })
+      });
```

Comment:
fine to just return results here since this forEach loop will be in handleSearchResults() since we will need to handle the searchByMap() results the same exact way